src/g3log/logmessage.hpp

```diff
@@ -87,7 +88,8 @@ namespace g3 {
       // Complete access to the raw data in case the helper functions above
       // are not enough.
       //
-      std::time_t _timestamp;
+      // TODO: unique_ptr should be considered instead of shared_ptr
+      std::shared_ptr<Timestamp> _timestamp;
```

Comment:
Actually you gave a great idea, but it is not exactly what you explained :)

But first I did a bit of analysis using templates for timestamps. Even though it is possible however it would be a HUGE g3log change. Using templates will affect pretty much all the layers: macros, LogCapture, LogMessage, moving most of the implementation code to headers... In other words it is not feasible.

This conclusion left me with using pointers. I started thinking about "placement new()" feature which I have never used, so it would be cool to try it out. It is possible to allocate single memory chunk for log message that would contain log message and timestamp of whatever type using "placement new()" feature. So memory wise log message would be a single block which would prevent from cache misses. Application wise it would be a raw pointers used in log message objects to the timestamps with a bit of woodoo code while creating objects. I almost started playing with this approach but then you proposed using strings...

Frankly I think that using strings in log messages is not a good solution. It is not elegant, it requires reserving some predefined buffer in log message which limits formatting functionality and ... it is not elegant :) However timestamp provider can return a struct with fields year, month, day, hour, minute, sec, microsecond. This way timestamp instance (which by the way would be always of the fixed size which is good for HW) would be part of the log message without any templates, pointers, heap allocation...Formatting would take care of converting this to a string depending of given format on logger's thread.

Would you accept my changes based on this approach?